app/src/main/java/com/sensirion/smartgadget/view/device_management/ScanDeviceFragment.java

```diff
@@ -669,31 +669,29 @@ public void run() {
         }
     }
 
-    private void waitForServiceSynchronization(@NonNull final BleDevice device) {
-        byte tryNumber = 0;
-        while (!RHTHumigadgetSensorManager.getInstance().isDeviceReady(device)
-                && ++tryNumber < DEVICE_SYNCHRONIZATION_MAX_NUMBER_SYNCHRONIZATION_TRIES) {
-
-            RHTHumigadgetSensorManager.getInstance().synchronizeDeviceServices(device);
-            try {
-                Thread.sleep(DEVICE_SYNCHRONIZATION_TIMEOUT_MILLISECONDS);
-            } catch (@NonNull final InterruptedException ignored) {
-            }
-
-            if (RHTHumigadgetSensorManager.getInstance().isDeviceReady(device)) {
+    private void waitForRHTServiceSynchronization(@NonNull final BleDevice device) {
+        final RHTHumigadgetSensorManager sensorManager = RHTHumigadgetSensorManager.getInstance();
+        for (byte tryNumber = 0; tryNumber < DEVICE_SYNCHRONIZATION_MAX_NUMBER_SYNCHRONIZATION_TRIES; tryNumber += 1) {
```

Comment:
instead of +=1 use ++ let's not set new conventions in the middle of a project.
Also when there's not a particular reason for a specific numeric type, it's best to use integer. With byte we're screwed if we want to increase the constant > 255